hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -460,6 +487,17 @@ protected void preCommit(String instantTime, HoodieCommitMetadata metadata) {
         Option.of(metadata), config, txnManager.getLastCompletedTransactionOwner());
   }
 
+  @Override
+  protected void saveStatisticsInfo(List<String> touchFiles, String cols, String indexPath, String saveMode) {
+    if (touchFiles.isEmpty() || cols.isEmpty() || indexPath.isEmpty()) {
+      LOG.warn("save nothing to index table");
+      return;
+    }
+    HoodieSparkEngineContext sparkEngineContext = (HoodieSparkEngineContext)context;
+    Zoptimize$.MODULE$.saveStatisticsInfo(sparkEngineContext
```

Comment:
not for log file.  since when the zorder/hilbert optimize has been done,  old files(include log file if the table is mor table) will replaced by new parquet file, so i donot consider log files.     but i agree with you that   it is better to build better abstractions to make it easily extend to future usecases